Makefile

```diff
@@ -0,0 +1,165 @@
+DOCKER_COMPOSE?=docker-compose
+EXEC?=$(DOCKER_COMPOSE) exec app
+CONSOLE=bin/console
+PHPCSFIXER?=$(EXEC) php -d memory_limit=1024m vendor/bin/php-cs-fixer
+
+.DEFAULT_GOAL := help
+.PHONY: help start stop restart install uninstall reset clear-cache tty clear clean
+.PHONY: db-diff db-migrate db-rollback db-reset db-validate wait-for-db
+.PHONY: watch assets assets-build
+.PHONY: tests lint lint-symfony lint-yaml lint-twig lint-twig php-cs php-cs-fix security-check test-schema test-all
+.PHONY: build up perm
+.PHONY: docker-compose.override.yml
+
+help:
+	@grep -E '(^[a-zA-Z_-]+:.*?##.*$$)|(^##)' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[32m%-30s\033[0m %s\n", $$1, $$2}' | sed -e 's/\[32m##/[33m/'
+
+##
+## Project setup
+##---------------------------------------------------------------------------
+
+start:                                                                                                 ## Start docker containers
+	$(DOCKER_COMPOSE) start
+
+stop:                                                                                                  ## Stop docker containers
+	$(DOCKER_COMPOSE) stop
+
+restart:                                                                                               ## Restart docker containers
+	$(DOCKER_COMPOSE) restart
+
+install: docker-compose.override.yml build up vendor perm                                                ## Create and start docker containers
+
+uninstall: stop                                                                                        ## Remove docker containers
+	$(DOCKER_COMPOSE) rm -vf
+
+reset: uninstall install                                                                               ## Remove and re-create docker containers
+
+clear-cache: perm
+	$(EXEC) $(CONSOLE) cache:clear --no-warmup
+	$(EXEC) $(CONSOLE) cache:warmup
+
+tty:                                                                                                   ## Run app container in interactive mode
+	$(EXEC) /bin/bash
+
+clear: perm                                                                                            ## Remove all the cache, the logs, the sessions and the built assets
+	$(EXEC) rm -rf var/cache/*
+	$(EXEC) $(CONSOLE) redis:flushall -n
+	rm -rf var/log/*
+	rm -rf public/build
+	rm -f var/.php_cs.cache
+
+clean: clear                                                                                           ## Clear and remove dependencies
+	rm -rf vendor node_modules
+
+
+##
+## Database
+##---------------------------------------------------------------------------
+
+wait-for-db:
+	$(EXEC) php -r "set_time_limit(60);for(;;){if(@fsockopen('db',3306)){break;}echo \"Waiting for MySQL\n\";sleep(1);}"
+
+db-diff: vendor wait-for-db                                                                            ## Generate a migration by comparing your current database to your mapping information
```

Comment:
So, If we don't use the MySql all these command will wait 60 seconds?

Reply:
Yes, in fact... I can remove all the db part, or just remove the waiting part and the db-reset part.

Reply:
This is actually quite nice oneliner. Wait until services are ready is very useful, so I would like to see something like this in the makefile, even if commented out as we use sqlite here.

Reply:
Yes, but it's difficult to add it in the Makefile whithout MySQL... Basically I've add all the Makefile db part in the case the user would like to use MySQL, but then we've removed all MySQL trace...